slkspec/core.py

```diff
@@ -135,23 +155,686 @@ def name(self) -> str:
+        # +==========================================================
+        def _start_recalls() -> None:
+            nonlocal file_ids_multiple_tapes
+            nonlocal tapes_success
+            nonlocal multi_tape_files_success
+            nonlocal tapes_active
+            nonlocal multi_tape_files_active
+            nonlocal tapes_failed
+            nonlocal multi_tape_files_failed
+            nonlocal job_tape_mapping
+            nonlocal job_multi_tape_file_mapping
+            nonlocal tape_job_mapping
+            nonlocal multi_tape_file_job_mapping
+            nonlocal all_tapes_done
+            nonlocal all_multi_tape_files_done
+            nonlocal tapes
+            nonlocal tape_file_mapping
+            file_ids: list[int]
+            job_ids_to_be_removed: set[int]
+            job_id: int
+            logger.debug("Recall function started")
+            msg: str
+            job_status: pyslk.base.stati.StatusJob
+
+            # +----------------------------------------------------------
+            # | CHECK IF WE NEED TO RUN THIS FUNCTION
+            # +----------------------------------------------------------
+            # all_tapes_done is not set => check if conditions to set this are fulfilled => print user info once
+            if (
+                len(
+                    [
+                        tape
+                        for tape in tapes
+                        if tape not in tapes_success and tape not in tapes_failed
+                    ]
+                )
+                == 0
+                and not all_tapes_done
+            ):
+                all_tapes_done = True
+                logger.info("All tapes have been processed.")
+
+            # all_multi_tape_files_done is not set => check if conditions to set this are fulfilled
+            #   => print user info once
+            if (
+                len(
+                    [
+                        file_id
+                        for file_id in file_ids_multiple_tapes
+                        if file_id not in multi_tape_files_success
+                        and file_id not in multi_tape_files_failed
+                    ]
+                )
+                == 0
+                and not all_multi_tape_files_done
+            ):
+                all_multi_tape_files_done = True
+                logger.info(
+                    "All files split amongst multiple tapes have been processed."
+                )
+
+            # leave directly if all tapes and all multi-tape-files have been processed
+            if all_tapes_done and all_multi_tape_files_done:
+                logger.debug("Recall function has nothing to do; leaving")
+                return
+
+            # +----------------------------------------------------------
+            # | CHECK STATUS OF RUNNING JOBS
+            # +----------------------------------------------------------
+            # check if there are jobs running for whole tapes:
```

Comment:
I am not a fan of these comment, ideally this would become obvious from the function names.

Reply:
I'll check it.

Reply:
Removed